lib/modules/@apostrophecms/piece-type/src/apos/components/AposPiecesManager.vue

```diff
@@ -0,0 +1,258 @@
+          </div>
+        </template>
+      </AposModalBody>
+      <!-- TODO: Trigger the piecesEditor another way. -->
+      <component
+        v-if="inserting" :module-name="moduleName"
+        :is="options.components.insertModal" @close="inserting = false"
+        @saved="finishSaved"
+      />
+    </template>
+  </AposModal>
+</template>
+
+<script>
+import AposTableMixin from '../../../../modals/mixins/AposTableMixin';
+
+export default {
+  name: 'AposPiecesManager',
+  mixins: [ AposTableMixin ],
+  props: {
+    moduleName: {
+      type: String,
+      required: true
+    }
+  },
+  data() {
+    return {
+      modal: {
+        active: false,
+        type: 'overlay',
+        showModal: false
+      },
+      pieces: [],
+      lastSelected: null,
+      totalPages: 1, // TODO: Make use of these.
+      currentPage: 1, // TODO: Make use of these.
+      filterValues: {}, // TODO: Make use of these.
+      inserting: false // TODO: Make use of these.
+    };
+  },
+  computed: {
+    options() {
+      return window.apos.modules[this.moduleName];
+    },
+    moduleLabels() {
+      console.info(this.options);
+      return {
+        singular: this.options.label,
+        plural: this.options.pluralLabel
+      };
+    },
+    moduleTitle () {
+      return `Manage ${this.moduleLabels.plural}`;
+    },
+    // headers:
+    rows() {
+      const rows = [];
+      if (!this.pieces || !this.headers.length) {
+        return [];
+      }
+
+      this.pieces.forEach(piece => {
+        const data = {};
+
+        this.headers.forEach(column => {
+          data[column.name] = piece[column.name];
+        });
+        rows.push(data);
+      });
+
+      return rows;
+    },
+    emptyDisplay() {
+      return {
+        title: `No ${this.moduleLabels.plural || this.moduleLabels.singular} Found`,
+        message: '',
+        emoji: '📄'
+      };
+    },
+    selectAllState() {
+      if (this.selectAllValue.data.length && !this.selectAllChoice.indeterminate) {
+        return 'checked';
+      }
+      if (this.selectAllValue.data.length && this.selectAllChoice.indeterminate) {
+        return 'indeterminate';
+      }
+      return 'empty';
+    }
+  },
+  // TODO: Work these back into the toolbar.
+  // watch: {
+  //   filterValues: {
+  //     deep: true,
+  //     handler() {
+  //       this.update();
+  //     }
+  //   },
+  //   currentPage() {
+  //     this.update();
+  //   }
+  // },
+  // created() {
+  //   this.options.filters.forEach(filter => {
+  //     this.filterValues[filter.name] = filter.choices[0].value;
+  //   });
+  // },
+  async mounted() {
+    // Get the data. This will be more complex in actuality.
+    this.modal.active = true;
+    this.getPieces();
+  },
+  methods: {
+    cancel() {
+      this.modal.showModal = false;
+    },
+    async finishSaved() {
+      await this.getPieces();
+
+      this.inserting = false;
+    },
+    async getPieces () {
+      apos.bus.$emit('busy', true);
```

Comment:
I believe @falkodev was working on a `busy` option to `apos.http.get` and friends which would take care of emitting the busy true / busy false events for you, saving a lot of try/finally calls and possible bugs.